src/System Application/App/File System/src/Account/FileAccountImpl.Codeunit.al

```diff
@@ -0,0 +1,227 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.FileSystem;
+
+using System.Utilities;
+using System.Text;
+
+codeunit 9451 "File Account Impl."
+{
+    Access = Internal;
+    InherentPermissions = X;
+    InherentEntitlements = X;
+    Permissions = tabledata "File System Connector Logo" = rimd,
+                  tabledata "File Scenario" = imd;
+
+    procedure GetAllAccounts(LoadLogos: Boolean; var TempFileAccount: Record "File Account" temporary)
+    var
+        FileAccounts: Record "File Account";
+        FileSystemConnector: Interface "File System Connector";
+        Connector: Enum "File System Connector";
+    begin
+        TempFileAccount.Reset();
+        TempFileAccount.DeleteAll();
+
+        foreach Connector in Connector.Ordinals do begin
+            FileSystemConnector := Connector;
+
+            FileAccounts.DeleteAll();
+            FileSystemConnector.GetAccounts(FileAccounts);
+
+            if FileAccounts.FindSet() then
+                repeat
+                    TempFileAccount := FileAccounts;
+                    TempFileAccount.Connector := Connector;
+
+                    if LoadLogos then
+                        ImportLogo(TempFileAccount, Connector);
+
+                    if not TempFileAccount.Insert() then;
```

Comment:
I have used the Email Module as a foundation. Not because I'm lazy, but because I think it makes sense to follow identical approaches if they have already proven themselves somewhere.

But in this case you are right, it makes no sense at all, it can't be twice in the temp record...